arch/arm/core/aarch32/cortex_r/reset.S

```diff
@@ -26,77 +27,161 @@ GDATA(z_arm_sys_stack)
 GDATA(z_arm_fiq_stack)
 GDATA(z_arm_abort_stack)
 GDATA(z_arm_undef_stack)
-
-#define STACK_MARGIN	4
-
+#if defined(CONFIG_PLATFORM_SPECIFIC_INIT)
+GTEXT(z_platform_init)
+#endif
 
 /**
  *
  * @brief Reset vector
  *
- * Ran when the system comes out of reset. The processor is in thread mode with
- * privileged level. At this point, the main stack pointer (MSP) is already
- * pointing to a valid area in SRAM.
+ * Ran when the system comes out of reset. The processor is in Supervisor mode
```

Comment:
That's actually accurate - there is no execution mode named "thread" in Cortex-R :)